src/Raven.Server/Documents/Document.cs

```diff
@@ -13,6 +14,108 @@ public class TimeSeriesStream
         public IEnumerable<DynamicJsonValue> TimeSeries;
         public string Key;
     }
+
+    public sealed class QueriedDocument : Document, IResetSupport<List<QueriedDocument>>
+    {
+        private static readonly QueriedDocument Empty = new();
+        
+        private int _refCount = 1;
+        
+        
+        private QueriedDocument _parent;
+        private QueriedDocument _child;
+        private List<QueriedDocument> _childs;
+        public bool CanDispose => _refCount <= 1;
```

Comment:
Can it be 0 or negative? Is it valid then? Should we have debug assert for that?